Plutarch/List.hs

```diff
@@ -133,6 +138,20 @@ pconvertLists = phoistAcyclic $
       (\x xs -> pcons # x #$ self # xs)
       pnil
 
+-- | Extract head and tail of the list, throws error if list is empty.
+punsafeUncons :: 
+  PIsListLike list a =>
+  Term s (list a) ->
+  Term s (PPair a (list a))
+punsafeUncons = pelimList (\x -> pcon . PPair x) perror
```

Comment:
This should also be hoisted and a Plutarch-level function.

Reply:
What exactly to hoist here?